### DISCUSSION TITLE
MarketURL for horse racing

### DISCUSSION THREAD

**Discussion by enrico0506** [cloudbet/docs#97] [General]
Does some one know the correct full marketUrl for horse racing when betting on SP. I only know that the start of the marketUrl is: horse_racing.win_sp

**Comment by kgravenreuth** [comment]
Hi @enrico0506 ,

As mentioned in our [Cloudbet Trading API documentation](https://www.cloudbet.com/api/?urls.primaryName=Trading#/Trading/PlaceBet):

```md
Market URL, is compiled from the feed data as marketKey/outcome?params, if no params (empty string) are present omit the query string and format as marketKey/outcome
```

The `Winner (SP)` market is defined in our [Cloudbet API Markets Github gist](https://gist.github.com/kgravenreuth/6703e1e213aecac4d5728f2f699d34e7#file-markets-json-L4358-L4366)

```json
	"horse_racing.win_sp": {
		"Name": "Winner (SP)",
		"Description": "The starting price (SP) will be the official odds declared by the on-course bookmakers at the start of the race.",
		"Variables": [],
		"Split": "period=default",
		"Outcomes": {
			"dynamic.player": "{{player}}"
		}
	},
```

For example, if a `selection` has the field `outcome: "player=cd261e-annsam"`, then you can construct the `marketUrl` as `horse_racing.win_sp/player=cd261e-annsam`

For further details, you can also check out our [Sports Betting API Tutorial With Golang](https://www.cloudbet.com/en/blog/posts/sports-betting-api-tutorial-with-golang?utm_source=github.com&utm_medium=community&utm_campaign=axis) blog article. 

Hope this helps!

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by codelancers-sd** [reply]
Thanks Klaus! 
Could you help me out on how to formulate this marketurl?
`	"basketball.player_assists": {
		"Name": "{{player}} (Assists)",
		"Variables": [
			"player",
			"total"
		],
		"Split": "player={{player}}",
		"Outcomes": {
			"over": "Over {{total}}",
			"under": "Under {{total}}"
		}`

Is it something like this? This did not work for me though:
marketUrl = basketball.player_assists/player=playerid12345&over?total=17.5

Thanks!

**Reply by kgravenreuth** [reply]
Hi @codelancers-sd ,
The `marketUrl` in this case would be `basketball.player_assists/over?total=17.5&player=playerid12345`

However, please make sure that this market is actually offered in our Feed API response before making the bet. You can check this via the `marketKey`, in this case `basketball.player_assists`

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet
